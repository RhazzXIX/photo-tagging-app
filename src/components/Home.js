import { Link, Outlet } from "react-router-dom";
import ghLogo from "../assets/gitW.png";
import "../styles/Home.css";

export default function Home({ version }) {
  let homeUrlRef = "/photo-tagging-app";
  if (version === "kids") homeUrlRef = "/photo-tagging-app/kids";
  return (
    <>
      <header>
        <Link to={homeUrlRef}>Home</Link>
        <Link to="leaderboard">Leaderboard</Link>
      </header>
      <Outlet />
      <footer>
        <a href="https://github.com/RhazzXIX">
          <img src={ghLogo} alt="GitHub" />
        </a>
      </footer>
    </>
  );
}
