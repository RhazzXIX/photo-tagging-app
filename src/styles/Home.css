div#root {
  height: 100vh;
  width: 100vw;
  display: flex;
  flex-direction: column;
}

div#root > header,
div#root > footer {
  background-color: black;
  display: flex;
  justify-content: space-evenly;
  align-items: center;
  height: 52px;
}

div#root > header
a {
  text-decoration: none;
  color: white;
  font-family: 'Times New Roman', Times, serif;
  font-size: 1.3rem;
  border-radius: 8px;
  padding: 4px 8px;
}

div#root > header
a:hover {
  border: 1px solid white;
}

div#root > header
a:active {
  color: black;
  background-color: white;
}

div#root > footer
img:hover {
  filter: drop-shadow(0 0 4px white)
}
