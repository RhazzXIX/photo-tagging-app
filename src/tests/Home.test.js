import '@testing-library/jest-dom'
import { screen, render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import Home from '../components/Home'
import { BrowserRouter } from 'react-router-dom'

describe('Home component', () => {
  it('Renders on screen', () => {
    const { container } = render(
    <BrowserRouter>
      <Home />
    </BrowserRouter>    
    )
    expect(container).toMatchSnapshot()
  })
})