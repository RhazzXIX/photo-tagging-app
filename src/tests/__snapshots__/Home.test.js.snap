// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component Renders on screen 1`] = `
<div>
  <header>
    <a
      href="/photo-tagging-app"
    >
      Home
    </a>
    <a
      href="/leaderboard"
    >
      Leaderboard
    </a>
  </header>
  <footer>
    <a
      href="https://github.com/RhazzXIX"
    >
      <img
        alt="GitHub"
        src="gitW.png"
      />
    </a>
  </footer>
</div>
`;
